rcl_action/src/rcl_action/names.c

```diff
@@ -0,0 +1,112 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef __cplusplus
+extern "C"
```

Comment:
This is important for the headers, but I don't think it is needed in a `.c` file since it can't be included by C++ code.

Reply:
I noticed that this is in every `.c` file in `rcl`. I'm not sure why though.

Reply:
That's me, if you don't do that then when you change the signature in the header it doesn't complain that the signature in the .c file is different, because it can be interpreted as C++ and therefore does overloading.

I believe some compilers (like Win32) always default to C++ if possible, even for `.c` files, but I could be wrong about that.